app/src/main/java/com/github/se/icebreakrr/model/profile/ProfilesViewModel.kt

```diff
@@ -0,0 +1,148 @@
+package com.github.se.icebreakrr.model.profile
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.google.firebase.Firebase
+import com.google.firebase.firestore.GeoPoint
+import com.google.firebase.firestore.firestore
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+open class ProfilesViewModel(private val repository: ProfilesRepository) : ViewModel() {
+
+  private val _profiles = MutableStateFlow<List<Profile>>(emptyList())
+  val profiles: StateFlow<List<Profile>> = _profiles
+
+  private val _selectedProfile = MutableStateFlow<Profile?>(null)
+  val selectedProfile: StateFlow<Profile?> = _selectedProfile
+
+  private val _loading = MutableStateFlow(false)
+  val loading: StateFlow<Boolean> = _loading
+
+  private val _error = MutableStateFlow<Exception?>(null)
+  val error: StateFlow<Exception?> = _error
+
+  init {
+    repository.init { // TODO getProfilesInRadius}
+    }
+  }
+
+  companion object {
+    val Factory: ViewModelProvider.Factory =
+        object : ViewModelProvider.Factory {
+          @Suppress("UNCHECKED_CAST")
+          override fun <T : ViewModel> create(modelClass: Class<T>): T {
+            return ProfilesViewModel(ProfilesRepositoryFirestore(Firebase.firestore)) as T
+          }
+        }
+  }
+
+  /**
+   * Fetches profiles within a given radius around a center point with optional filters.
+   *
+   * @param center The center location as a GeoPoint.
+   * @param radiusInMeters The radius around the center in meters to search for profiles.
+   * @param gender Optional filter for gender.
+   * @param ageRange Optional filter for age range.
+   * @param tags Optional filter for specific tags.
+   */
+  fun getFilteredProfilesInRadius(
+      center: GeoPoint,
+      radiusInMeters: Double,
+      gender: Gender? = null,
```

Comment:
You can have several genders preferences, so I think this should be a List<Gender> rather than a single Gender

Reply:
Good catch, I will correct this